tools/topology/topology2/include/pipelines/cavs/host-src-capture.conf

```diff
@@ -0,0 +1,84 @@
+#
+# Host src capture pipeline
+#
+# A simple host src pipeline for capture. All attributes defined herein are namespaced
+# by alsatplg to "Object.Pipeline.host-src-capture.N.attribute_name".
+#
+# Usage: host-src-capture pipeline object can be instantiated as:
+#
+# Object.Pipeline.host-src-capture."N" {
+# 	period		1000
+# 	time_domain	"timer"
+# 	channels	2
+# 	rate		48000
+# }
+#
+# Where N is the unique pipeline ID within the same alsaconf node.
+#
+
+<include/common/input_audio_format.conf>
+<include/common/output_audio_format.conf>
+<include/components/copier.conf>
+<include/components/pipeline.conf>
+<include/components/src.conf>
+
+Class.Pipeline."host-src-capture" {
+
+	DefineAttribute."index" {}
+
+	<include/pipelines/pipeline-common.conf>
+
+	attributes {
+		!constructor [
+			"index"
+		]
+
+		!immutable [
+			"direction"
+		]
+
+		#
+		# host-src-capture objects instantiated within the same alsaconf node must have
+		# unique pipeline_id attribute
+		#
+		unique	"instance"
+	}
+
+	Object.Widget {
+		copier."1" {
+			copier_type	"host"
+			type	"aif_out"
+			node_type $HDA_HOST_INPUT_CLASS
+
+			<include/components/src_passthrough_format.conf>
+		}
+
+		src."1" {
+			rate_in 48000
+
+			<include/components/src_format_host_capture.conf>
+		}
+
+		pipeline."1" {
+			priority	0
+			lp_mode		0
+		}
+	}
+
+	Object.Base {
+		route.1 {
+			source	src..1
+			sink 	copier.host..1
```

Comment:
Yep, I'll try it. It would simplify a lot this kind of changes.

Reply:
I wonder how to avoid making a duplicate of host-gateway-capture.conf --> host-gateway-multirate-capture.conf with just the pass-through 8 - 192 kHz rates for host copier. The original supports also 4ch. I didn't add those because there's already 42 formats with stereo.

Reply:
@singalsu you can simply use host-gateway-capture.conf with the passthrough rates included inside the host copier object. It will override the audio formats that are in the class definition

Reply:
I've now changed the structure, hope it's now like you meant @plbossart and @ranj063 .